src/core/lint-hrefs-ids.js

```diff
@@ -0,0 +1,21 @@
+// Module core/lint-hrefs-ids
+// Warns about href's that do not have corresponding id's
+
+import { pub } from "core/pubsubhub";
+
+export const name = "core/lint-hrefs-ids";
+
+export function run(conf, doc, cb) {
+  console.log()
+  Array.from(doc.querySelectorAll('a[href]'))
+    .map(a => ({ a, href: a.getAttribute('href') }))
+    .filter(({ href }) => Boolean(href))
```

Comment:
combine the two filters into a single filter here. Also, do the filtering before the map, otherwise, we end up creating a bunch of extra objects that we don't need.